source/ns_cmdline.c

```diff
@@ -77,7 +77,10 @@
 #define TAB 0x09
 #define CAN 0x18
 
+#ifndef MAX_LINE
 #define MAX_LINE 2000
+#endif
```

Comment:
Could you please change the macro to something more specific for this library, like perhaps `MBED_CLIENT_CLI_MAX_LINE`?

Odds of "MAX_LINE" clashing with another define don't seem huge, but might as well avoid even the small chance.